src/application/look-up-wallet-address/look-up-wallet-address.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from 'react';
+import { useKey } from 'react-use';
+import { useForm } from 'react-hook-form';
+
+import { useCommandQuery } from 'shared/messaging';
+import { Closable, IconButton, Spinner } from 'shared/ui';
+import { useExtensionSettings } from 'shared/extension';
+
+import { GetResolvedAddressCommand, GetTwitterIdsCommand } from './commands';
+import { AddressList } from './address-list';
+
+export const LookUpWalletAddress = () => {
+  const [username, setUsername] = useState('');
+  const [isVisible, setIsVisible] = useState(false);
+
+  const { extensionSettings } = useExtensionSettings();
+
+  const isEnabled =
+    extensionSettings['entire-extension-enabled'] &&
+    extensionSettings['wallet-lookup-enabled'];
+
+  const twitterIdsQuery = useCommandQuery({
+    command: new GetTwitterIdsCommand({
+      username,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+  });
+
+  const addressesQuery = useCommandQuery({
+    command: new GetResolvedAddressCommand({
+      identifier: username,
+      twitterId: twitterIdsQuery.data?.twitterIDs
+        ? Object.values(twitterIdsQuery.data?.twitterIDs)[0]
+        : undefined,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+    enabled: username.length >= 3,
+  });
+
+  const form = useForm();
+
+  const userIdentifier = form.watch('user_identifier');
+
+  const onSubmit = () => {
+    const values = form.getValues();
+    setUsername(values.user_identifier);
+  };
+
+  const closeWidget = () => {
+    setIsVisible(false);
+    setUsername('');
+    form.reset();
+  };
+
+  useKey(
+    (event) => {
+      return event.ctrlKey && event.key === 'i';
+    },
+    () => {
+      return setIsVisible((previous) => {
+        return !previous;
+      });
+    },
+  );
+
+  if (!isVisible || !isEnabled) {
+    return null;
+  }
+
+  return (
+    <Closable
+      className="fixed right-6 top-6 z-[9990] p-0"
+      onClose={closeWidget}
+    >
+      <div className="w-[470px] rounded-md bg-white px-2 py-3">
+        <div className="relative">
+          <form onSubmit={form.handleSubmit(onSubmit)}>
+            <label
+              htmlFor="user_identifier"
+              className="mb-2 block text-base font-bold text-gray-700"
+            >
+              Look up an address
+            </label>
+            <div className="flex flex-row gap-x-[10px]">
+              <div className="relative grow">
+                <input
+                  autoFocus
+                  id="user_identifier"
+                  className="mb-1.5 box-border block h-[38px] w-full rounded border border-[#cccccc] bg-white py-2 pl-3 pr-10 align-middle font-sans text-sm text-[#333333] outline-none"
+                  placeholder="hello@idriss.xyz  |  +1 650...  |  @IDriss_xyz"
+                  {...form.register('user_identifier')}
+                />
```

Comment:
if you will try to find `register(` in repo you will only find this1, that's because in other places I use `Controller` component which is more flexible:
```tsx
<Controller
  render={({ field }) => {
   return <input {...field}> 
  }
 >
```

`field` under the hood includes things like `value`, `onChange`, etc.